pkg/kubectl/cmd/util/factory.go

```diff
@@ -336,6 +337,10 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			case autoscaling.GroupName:
 				return c.AutoscalingClient.RESTClient, nil
 			case batch.GroupName:
+				// XXX mapping is correct (batch/v2alpha1), the returned client is using batch/v1.
```

Comment:
From what @lavalamp pointed me to we need to wait for @krousey's changes to kubectl. I'm guessing it's about #25546 and #25676.